### ISSUE TITLE
Vague errors hide underlying issues

### ISSUE THREAD

**Issue by fizyk** [magefile/mage#357]
I had an issue where I used a library in one project's, I used in other, but I forgot to put it to go.mod.

All mage did was print this error:

```
Error determining list of magefiles: failed to list mage gofiles: exit status 1: 
```
Which is pertty vague.

**Comment by natefinch** [comment]
Yeah, we can do better than that. We should be able to expose the underlying compile error.
